torch/nn/modules/transformer.py

```diff
@@ -375,6 +375,100 @@ def forward(self, tgt, memory, tgt_mask=None, memory_mask=None,
         return tgt
 
 
+class MultiheadAttentionInProjection(Module):
+    r"""
+    Args:
+        in_embed_dim (int): Input embedding dimension
+        num_heads (int): Number of parallel attention heads.
+        out_embed_dim (int, optional): Output embedding dimension. If not
+            provided, then set to `in_embed_dim`. Must be divisible by
+            `num_heads`.
+    """
+    def __init__(self, in_embed_dim, num_heads, out_embed_dim=None):
```

Comment:
I'm starting to think it might be best to make a single module that does `q`, `k` and `v` all at once. It reduces flexibility, but it will be a more accessible interface. People that want the flexibility could use the `functional` as well to construct their own module. Thoughts on a split approach?

Reply:
Actually I don't have a strong opinion against this. @fmassa Do you have a preference?